terraform_token.go

```diff
@@ -71,20 +49,40 @@ func createTeamToken(ctx context.Context, c *client, teamID string) (*terraformT
 		return nil, err
 	}
 
-	tfToken := &terraformToken{}
-	tfToken.translateTeamToken(token)
-	return tfToken, nil
+	return &terraformToken{
+		ID:          token.ID,
+		Description: token.Description,
+		Token:       token.Token,
+	}, nil
+}
+
+func createUserToken(ctx context.Context, c *client, userID string) (*terraformToken, error) {
+	token, err := c.UserTokens.Generate(ctx, userID, tfe.UserTokenGenerateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return &terraformToken{
+		ID:          token.ID,
+		Description: token.Description,
+		Token:       token.Token,
+	}, nil
 }
 
 func (b *tfBackend) terraformTokenRevoke(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
-	tokenIDRaw, ok := req.Secret.InternalData["token_id"]
-	if !ok {
-		return nil, fmt.Errorf("secret is missing token id internal data")
+	client, err := b.getClient(ctx, req.Storage)
+	if err != nil {
+		return nil, fmt.Errorf("error getting client: %w", err)
 	}
 
-	tokenID, ok := tokenIDRaw.(string)
-	if !ok {
-		return nil, fmt.Errorf("secret is missing token id internal data")
+	teamID := ""
+	teamIDRaw, ok := req.Secret.InternalData["team_id"]
```

Comment:
From a top-down logic, should we perform the org check first?

Reply:
I swapped the ordering in fb738da, we now populate `organization` first. I believe that's what you are suggesting?